crates/subspace-farmer/src/dsn.rs

```diff
@@ -50,101 +88,75 @@ where
     DSN: DSNSync + Send + Sized,
     DSN::Stream: Unpin + Send,
     // On pieces might `break` with some result from the sync or just continue
-    OP: FnMut(FlatPieces, Vec<PieceIndex>) -> ControlFlow<anyhow::Result<()>> + Send + 'static,
+    OP: FnMut(FlatPieces, Vec<PieceIndex>) -> anyhow::Result<()> + Send + 'static,
 {
     let SyncOptions {
+        max_plot_size,
+        total_pieces,
         range_size,
         address,
     } = options;
-    let mut increasing_cursor = PieceIndexHashNumber::from(Sha256Hash::from(address));
-    let mut decreasing_cursor = increasing_cursor;
+    let address = PieceIndexHashNumber::from(Sha256Hash::from(address));
+    let max_distance_one_direction = if total_pieces < max_plot_size {
+        PieceIndexHashNumber::MAX / 2
+    } else {
+        PieceIndexHashNumber::MAX / total_pieces * max_plot_size / 2
+    };
+    let (from, is_underflow) = address.overflowing_sub(max_distance_one_direction);
+    let (end, is_overflow) = address.overflowing_add(max_distance_one_direction);
```

Comment:
Here is what I mean by explaining things in plain English:
* we have a public key on the ring
* we have a specific sector size with public key in the center that farmer cares about
* we need to compute the start and end of that sector

Translated into code it becomes this:
```suggestion
    let sync_sector_size = (PieceIndexHashNumber::MAX / total_pieces).saturating_mul(max_plot_size);
    let from = public_key.wrapping_sub(sync_sector_size / 2);
    let to = public_key.wrapping_add(sync_sector_size / 2);
```

Looks easier to understand to me at least and doesn't even require a comment, while you can explain it clearly with simple words. By comparing `from` and `end` you can identify whether it wrapped around `0` or not.

Also naming is important with similar variables: if one is `from` then another one is `to`, if one is `start` then another will be `end`. Not `from` and `end` or `start` and `to`.